README.md
<p align="center">
  <img src="https://github.com/jbkuczma/NBAreact/blob/master/media/swish_app.png">
</p>

![](https://github.com/jbkuczma/NBAreact/blob/master/media/swish_github.png)

<p align="center">
  <a href="">
     <img src="https://img.shields.io/badge/%F0%9F%8F%80-This%20Is%20Why%20We%20Play-blue.svg?&style=flat-square" alt="This Is Why We Play">
  </a>
  <a href="https://github.com/jbkuczma/NBAreact/issues">
    <img src="https://img.shields.io/github/issues/jbkuczma/NBAreact.svg?style=flat-square" alt="GitHub issues">
  </a>
</p>
<p align="center">
  <a href="https://itunes.apple.com/us/app/swish-basketball-stats/id1361567861">
    <img alt="Download on the App Store" title="App Store" src="http://i.imgur.com/0n2zqHD.png" width="200">
  </a>
</p>

# Swish
An iOS and Android NBA app created with React Native. If you would like to request a feature, find a bug, have a question, or would like to leave feedback, open an [issue](https://github.com/jbkuczma/NBAreact/issues)! ⭐️ this repo to show support!

## Download
- iOS - Download [here](https://itunes.apple.com/us/app/swish-basketball-stats/id1361567861)!
- Android - Coming soon!

## Features
- Scores and stats for in-progress and completed games
- Detailed play by play
- Full team stats for a season
- Full player stats for a season and over their career
- Detailed player game stat graphs
- Conference rankings
- League leaders
- Videos and highlights (coming soon)
- Compare two players (coming soon)

## Libraries Used
[React Native](https://github.com/facebook/react-native)

[React Navigation](https://reactnavigation.org/)

[Redux](https://redux.js.org/)

[Redux Thunk](https://github.com/gaearon/redux-thunk)

[React-Native-SVG-Charts](https://github.com/JesperLekland/react-native-svg-charts)

[React-Native-Modal-Dropdown](https://github.com/sohobloo/react-native-modal-dropdown)

[nba.js](https://github.com/kshvmdn/nba.js) (custom and modified implementation to work with React Native)

## Resources Used
[Set Splash Screen iOS](https://medium.com/the-react-native-log/change-default-launch-screen-in-react-native-ios-app-544f94f1e947)

[Test your React Native App on your iPhone](https://medium.com/react-native-development/deploying-a-react-native-app-for-ios-pt-1-a79dfd15acb8)

[How to Submit to the App Store](https://clearbridgemobile.com/how-to-submit-an-app-to-the-app-store/)

# Getting Started

  - Fork (optional) and clone repo

        git clone https://github.com/jbkuczma/NBAreact.git && cd NBAreact

  - Install and link dependencies

        npm install

        npm link

  - Run

        npm run start:ios

        npm run start:iphoneX

        npm run start:android (an emulator must be running first)


CHANGELOG.md
# Swish Changelog

All notable changes to Swish will be documented here

# 2.5 (In development)

# 2.4 (5-27-2018)
- Tapping a tab button will take you to its initial screen (#55)
- Fix bug where the LeadTracker graph would not show overtime (#70)
- Player name column in a boxscore is now fixed; only stats are now scrollable (#72)
- Fixed bug where completed games were showing their tip-off time (#75)
- Add iPad support (#68)

# 2.3 (5-6-2018)

- Added a new screen that displays graphs for various player stats during a specific game (#49)
- Add refresh capabilities to the game detail screen (#63)
- Show a circle with the team's name and team color in a game cell instead of just the team (#66)
- Fixed a bug where clicking the date in the scores screen header would crash the app
- Fixed a bug where the boxscore wouldn scroll horizontally on iOS but not Android (#58)

# 2.2 (4-18-2018)

- Career stats for players added (#51)
- Past game scores for a team added (#61)
- Only allow the games list to be refreshed on current date (#45)

# 2.1 (3-30-2018)

- League leaders tab and page added (#41)
- Ability to search for any player added
- Matchup added to navbar title added
- PlayerScreen.js refractored to allow navigation from various screens (#54)
- Boxscore.js refractored

# 2.0 (3-14-2018)

- Complete rewrite and UI update (#33)
- Redux integrated (#33)
- App renamed from NBAreact to Swish (#38)

# pre 2.0

- See 0cac03d for status of the code base and app before the rewrite


App/NBAreact.js
import React, { Component } from 'react'
import { Text, View, Image } from 'react-native'
import { StackNavigator, TabNavigator, NavigationActions } from 'react-navigation'

import ScoresScreen from './components/ScoresScreen'
import GameDetailScreen from './components/GameDetailScreen'
import StandingsScreen from './components/StandingsScreen'
import TeamScreen from './components/TeamScreen'
import PlayerScreen from './components/PlayerScreen'
import PlayerGameDetailScreen from './components/PlayerGameDetailScreen'
import TeamStats from './components/GameDetailScreen/TeamStats'
import BoxScore from './components/GameDetailScreen/BoxScore'
import PlayByPlay from './components/GameDetailScreen/PlayByPlay'
import LeagueLeaders from './components/LeagueLeaders'

import SearchBar from './components/common/SearchBar'

const defaultNavigationOptions = {
  headerTintColor: '#D3D3D3',
  headerStyle: {
    backgroundColor: '#171717',
    borderBottomWidth: 0
  },
  headerTitleStyle: {
    fontSize: 18,
  }
}

const GameDetailNavigator = TabNavigator({
  'Team Stats': { screen: TeamStats },
  'Boxscore': { screen: BoxScore },
  'Play by Play': { screen: PlayByPlay }
}, {
  tabBarPosition: 'top',
  swipeEnabled: false,
  tabBarOptions: {
    activeTintColor: '#FFFFFF',
    inactiveTintColor: '#777777',
    inactiveBackgroundColor: '#151516',
    activeBackgroundColor: '#171717',
    showIcon: false,
    indicatorStyle: {
      borderBottomColor: '#F7971E',
      borderBottomWidth: 2,
    },
    labelStyle:{
      fontSize: 14,
      justifyContent: 'center',
      alignItems: 'center',
    },
    style:{
      backgroundColor: '#111111'
    },
    tabStyle: {
      justifyContent: 'center',
      alignItems: 'center',
    }
  },
})

const ScoresStack = StackNavigator({
  Home: {
    screen: ScoresScreen,
    navigationOptions: { header: null }
  },
  Game: {
    screen: GameDetailNavigator,
    navigationOptions: defaultNavigationOptions
  }
})

const StandingsStack = StackNavigator({
  Home: {
    screen: StandingsScreen,
    navigationOptions: { header: null }
  },
  Team: {
    screen: TeamScreen,
    navigationOptions: defaultNavigationOptions
  },
  Player: {
    screen: PlayerScreen,
    navigationOptions: defaultNavigationOptions
  },
  Game: {
    screen: GameDetailNavigator,
    navigationOptions: defaultNavigationOptions
  },
  PlayerGameDetail: {
    screen: PlayerGameDetailScreen,
    navigationOptions: defaultNavigationOptions
  }
})

const LeagueLeadersStack = StackNavigator({
  Home: {
    screen: LeagueLeaders,
    navigationOptions: {
      headerStyle: {
        // backgroundColor: '#111111'
      }
    }
  },
  Player: {
    screen: PlayerScreen,
    navigationOptions: defaultNavigationOptions
  },
  PlayerGameDetail: {
    screen: PlayerGameDetailScreen,
    navigationOptions: defaultNavigationOptions
  }
})

export default TabNavigator({
  Scores: { screen: ScoresStack },
  Standings: { screen: StandingsStack },
  Leaders: { screen: LeagueLeadersStack }
},{
  navigationOptions: ({ navigation }) => ({
    tabBarIcon: ({ focused, tintColor }) => {
      const { routeName } = navigation.state
      let iconName
      let iconStyle = { height: 42, width: 42 }
      if (routeName === 'Scores') {
        iconName = require('./Assets/icons/scoreboard.png')
      } else if (routeName === 'Standings') {
        iconName = require('./Assets/icons/trophy.png')
      } else if (routeName === 'Leaders') {
        iconName = require('./Assets/icons/bars.png')
        iconStyle = { height: 28, width: 28 }
      }

      return <Image source={iconName} style={[ iconStyle, { tintColor: tintColor }]} />
    },
    tabBarOnPress: ({ scene, jumpToIndex }) => {
      const { route, focused, index } = scene
      if (focused) {
        if (route.index > 0) {
          const { routeName, key } = route.routes[1]
          navigation.dispatch(
            NavigationActions.back({ key })
          )
        }
      } else {
        jumpToIndex(index)
      }
    },
  }),
  tabBarOptions: {
    activeTintColor: '#F7971E',
    inactiveTintColor: 'gray',
    style: {
      backgroundColor: '#171717',
    }
  },
  tabBarPosition: 'bottom'
})
